src/legacy/core_plugins/data/public/plugin.ts

```diff
@@ -85,7 +85,7 @@ export class DataPlugin implements Plugin<DataSetup, void, DataPluginSetupDepend
         indexPatterns: indexPatternsService.indexPatterns,
       }),
       query: this.query.setup(),
-      search: this.search.setup(),
+      search: this.search.setup(savedObjectsClient),
```

Comment:
nit:  Other services pass in objects

Reply:
@lizozom the search service setup method returns the savedQueryService that is an object, so in effect, we are passing in an object.

Reply:
@lizozom are you just saying you'd prefer this:

`this.search.setup({ savedObjectsClient })`

?